src/agoradatatools/etl/transform/rna_de_aggregate.py

```diff
@@ -0,0 +1,188 @@
+    "mouse_gene_metadata": ["ensembl_gene_id", "gene_symbol", "alias"],
+    "model_info": ["model", "matched_controls", "model_type"],
+    "biodom_genes_mm": [
+        "biodomain",
+        "abbr",
+        "label",
+        "color",
+        "go_id",
+        "goterm_name",
+        "n_symbol",
+        "symbol",
+        "ensembl_id",
+    ],
+}
+
+
+def _quick_validate_data_file(
+    file_name: str,
+    data_file: pd.DataFrame,
+    required_columns: List[str] = [
+        "ensembl_gene_id",
+        "log2FoldChange",
+        "padj",
+        "model",
+        "case",
+        "control",
+        "age",
+        "sex",
+        "tissue",
+    ],
+) -> None:
+    """
+    Quick validate the data file. Only validates the required columns.
+
+    Args:
+        file_name (str): The name of the file.
+        data_file (pd.DataFrame): The data file to validate.
+        required_columns (List[str]): The required columns to validate.
+
+    Raises:
+        ValueError: If the data file is empty or if the required columns are missing.
+
+    Returns:
+        None
+    """
+    if data_file.empty:
+        raise ValueError("Data file is empty")
+
+    # Validate required columns
+    missing_columns = [col for col in required_columns if col not in data_file.columns]
+    if missing_columns:
+        raise ValueError(
+            f"Missing required columns in {file_name} dataset: {', '.join(missing_columns)}. "
+            f"Please ensure the {file_name} dataset contains all required columns: {', '.join(required_columns)}."
+        )
+
+
+def transform_rna_de_aggregate(
+    datasets: Dict[str, pd.DataFrame],
+    required_input: Dict[str, List[str]] = REQUIRED_INPUT,
+) -> List[Dict[str, Any]]:
+    """
+    Transforms the rna_de_aggregate source files into a structured format for Model AD.
+    Groups by gene, model, tissue, and sex, with age-based entries containing log2_fc and adj_p_val.
+    """
+    check_required_datasets_and_columns(datasets, required_input)
+
+    # Pre-compute lookup dictionaries for efficient lookups
+    rnaseq_genotype_label_map_df = datasets["rnaseq_genotype_label_map"].fillna("")
+    mouse_gene_metadata_df = datasets["mouse_gene_metadata"].fillna("")
+    model_info_df = datasets["model_info"].fillna("")
+    biodom_genes_mm_df = datasets["biodom_genes_mm"].fillna("")
+
+    # Create lookup dictionaries
+    gene_metadata_dict = mouse_gene_metadata_df.set_index("ensembl_gene_id")[
+        "gene_symbol"
+    ].to_dict()
+    model_info_dict = model_info_df.set_index("model")["model_type"].to_dict()
+
+    # Create label map dictionaries for efficient lookups
+    label_map_dict = {}
+    model_group_dict = {}
+    for _, row in rnaseq_genotype_label_map_df.iterrows():
+        key = (row["model"], row["genotype"])
+        label_map_dict[key] = row["display_label"]
+        model_group_dict[row["model"]] = row["model_group"]
```

Comment:
You can shortcut this just like you did with the other dfs above:
```suggestion
    label_map_dict = rnaseq_genotype_label_map_df.set_index(["model", "genotype"])["display_label"].to_dict()
    model_group_dict = rnaseq_genotype_label_map_df.set_index(["model", "genotype"])["model_group"].to_dict()
```